src/main/java/org/sagebionetworks/bridge/udd/helper/ZipHelper.java

```diff
@@ -57,21 +63,48 @@ public void zip(List<File> fromList, File to) throws IOException {
 
     /**
      * Zips and encrypts the list of input files and writes the result to the output file.
-     * @param fromList
+     * @param filesToAdd
      *          list of input files
-     * @param to
+     * @param outputZipFile
      *          output file to write the zip file to
      * @param password
      *          given password to secure the file with
      * @throws IOException
      *          if reading from input or writing to output fails
      */
-    public void zipWithPassword(List<File> fromList, File to, String password) throws IOException {
-        ZipParameters zipParameters = new ZipParameters();
-        zipParameters.setEncryptFiles(true);
-        zipParameters.setEncryptionMethod(EncryptionMethod.AES); // AES 256 by default
+    public void zipWithPassword(List<File> filesToAdd, File outputZipFile, String password) throws IOException {
+        ZipParameters zipParameters = buildZipParameters();
+        byte[] buffer = new byte[BUFFER_SIZE];
+        int readLen;
+
+        try (OutputStream bufferedOutputStream = new BufferedOutputStream(fileHelper.getOutputStream(outputZipFile))) {
+            net.lingala.zip4j.io.outputstream.ZipOutputStream zipOutputStream =
+                    new net.lingala.zip4j.io.outputstream.ZipOutputStream(bufferedOutputStream, password.toCharArray());
+            for (File file : filesToAdd) {
+                zipParameters.setFileNameInZip(file.getName());
+                zipOutputStream.putNextEntry(zipParameters);
+
+                try (InputStream inputStream = fileHelper.getInputStream(file)) {
+                    while ((readLen = inputStream.read(buffer)) != -1) {
+                        zipOutputStream.write(buffer, 0, readLen);
+                    }
```

Comment:
Consider using https://guava.dev/releases/19.0/api/docs/com/google/common/io/ByteStreams.html#copy(java.io.InputStream,%20java.io.OutputStream) so that you don't have to do your own stream buffer management.